### DISCUSSION TITLE
Where is $PYENV_ROOT/bin?

### DISCUSSION THREAD

**Discussion by drewcassidy** [pyenv/pyenv#2684] [Q&A] [answered]
The readme setup includes lines like

```
set -Ux PYENV_ROOT $HOME/.pyenv
fish_add_path $PYENV_ROOT/bin
```

yet in other places its

`$(pyenv root)/shims:/usr/local/bin:/usr/bin:/bin`

I don't see a bin folder in my .pyenv folder, but I DO have a shims folder. Is this an error in the readme or am I missing something?

**Comment by native-api** [comment, marked as answer]
You won't have `$PYENV_ROOT/bin` if Pyenv is installed at a path different than `$PYENV_ROOT` -- e.g. with Homebrew.

That's why the suggested setup for Bash/Zsh only adds that directory to `PATH` if `pyenv` is not already on `PATH`.

We did not duplicate this feature in Fish setup because we're less familiar with its syntax. We'll gladly accept a PR fixing that.

**Reply by drewcassidy** [reply]
So `$PYENV_ROOT/bin` includes pyenv itself, while `$PYENV_ROOT/shims` contains all the python shims?

**Reply by native-api** [reply]
"Pyenv itself" constitutes what you can see in the code tree at https://github.com/pyenv/pyenv . In the course of its work, it also creates some directories under `$PYENV_ROOT` such as `shims` and `versions`.
